Core/src/Detector/DetectorVolume.cpp

```diff
@@ -103,12 +107,6 @@ bool Acts::Experimental::DetectorVolume::inside(const GeometryContext& gctx,
   if (not excludeInserts or m_volumes.external.empty()) {
     return true;
   }
-  // Check exclusion through subvolume
-  for (const auto v : volumes()) {
```

Comment:
Not sure I understand this change, is this not what we want? Is it not needed anymore?

Reply:
I didn't see a reason why we would put a hole into our volume where there are subvolumes... @asalzburger did you have a specific use case in mind? because for me it broke the lookup mechanism for the volume

Reply:
I see. I also refactored the lookup mechanism because it didn't give me the right volume if the hierarchy is too deep. it also looks to me like this `inside` method could break after 2 levels down the tree?

maybe we can discuss this tomorrow

Reply:
Absolutely, I think we can achieve both - a hierarchy based version and a unique assignment. 

For me the single requirement would be that you can unique associate - and thus let the portals do the guiding through the detector.

My morning is fully booked, I will find you as soon as I am done.